### PR TITLE
feat: hide UI depending on feature visibility flags

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Hide the UI component that is currently no supported depending on the simple boolean flags

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/util/debug/FeatureVisibilityFlags.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+//Conversation View → search icon
+//User Profile → Edit icon
+//long press on text → edit entry
+//long press on images → edit
+//long press on images → copy
+
+
+// Those flags can be removed once we set all flags to true :)
+object FeatureVisibilityFlags {
+    const val Settings = false
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/util/debug/FeatureVisibilityFlags.kt:29]: We're not supposed to remove ALL settings, right? Only the `App Settings` inside the `Settings` screen.

I'd rename this to `AppSettings`

Comment (vitorhugods) [reply app/src/main/kotlin/com/wire/android/util/debug/FeatureVisibilityFlags.kt:29]: Also, `BackupAndRestore` is missing.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1025/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4496471)(4496471)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4496471)(4496471)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/HomeDrawer.kt [hunk 2]
```diff
@@ -100,7 +98,14 @@ fun HomeDrawer(
 
         Spacer(modifier = Modifier.weight(1f))
 
-        val bottomItems = listOf(Settings, Support, Debug)
+        val bottomItems = buildList {
+            if (LocalFeatureVisibilityFlags.current.Settings) {
+                add(Settings)
+            }
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/HomeDrawer.kt:104]: As said in the other comment. It's not the whole Settings screen that is supposed to be hidden, right? It's just the `App Settings` section INSIDE the main `Settings` screen, and the `Backup and Restore` option too.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1025/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4496601)(4496601)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4496601)(4496601)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1025/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4497059)(4497059)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4497059)(4497059)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1025/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4497263)(4497263)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4497263)(4497263)-dev-release.apk)

**Pull request merged**